acados/ocp_nlp/ocp_nlp_cost_nls.c

```diff
@@ -286,6 +288,11 @@ int ocp_nlp_cost_nls_model_set(void *config_, void *dims_, void *model_,
         double *scaling_ptr = (double *) value_;
         model->scaling = *scaling_ptr;
     }
+    else if (!strcmp(field, "outer_hess_is_diag"))
+    {
+        int *outer_hess_is_diag_ptr = (int *) value_;
+        model->outer_hess_is_diag = *outer_hess_is_diag_ptr;
```

Comment:
I think that for `CONL` cost, the diagonal detection is pretty bullet-proof.
However, for `NLS` / `LS` cost, we allow updating `W` via `cost_set` in the interface.
I think, we should thus check in C, in this function here, if the `W` supplied is actually diagonal and then assign `outer_hess_is_diag` only internally.
In this case, we could actually remove setting it manually for NLS cost in the templates.